### ISSUE TITLE
Cannot use phpvirtualbox anymore after update

### ISSUE THREAD

**Issue by PatiB87** [phpvirtualbox/phpvirtualbox#219]
Hey,

I updated my NUC from Debian Stretch to Buster and therefore also updated VirtualBox - now running VirtualBox 6.0.14r133895
I was also able to start all the VMs via CLI and everything works.

Thing is: when I now access my PHPVirtualBox I only get this:

```
An unknown PHP error occurred. This is most likely a syntax error in config.php in phpVirtualBox's folder. The most common errors are an unclosed quote or a missing semicolon in a configuration item that has been entered (e.g. location, username, or password).

Depending on your PHP configuration, navigating directly to config.php in your web browser may display the PHP error message.

If find that this is not the case, or have no idea what this error message means, please raise the issue at http://sourceforge.net/p/phpvirtualbox/discussion/help/

```

What could be the issue? How can I help you debugging?
Do I have to reinstall the whole PHPVirtualBox?
If that could help, could you link me how to do this?

Cheers,
Patrick

**Comment by dumarcelino** [comment]
https://sourceforge.net/p/phpvirtualbox/bugs/56/

**Comment by dumarcelino** [comment]
https://www.howtoforge.com/tutorial/managing-a-headless-virtualbox-installation-with-phpvirtualbox-on-ubuntu-16.04/
I believe some libraries were missing from your installation.

Check out this install for ubuntu what your debian for php may be missing.


good luck ;-)


Eduardo Lima Marcelino

**Comment by PatiB87** [comment]
I tried starting vboxweb as user vbox manually and then connect with phpvirtualbox, I get this, which looks fine so far:
`vbox@Intel-NUC:/var/www/html/phpvirtualbox$ vboxwebsrv -H 127.0.0.1
Oracle VM VirtualBox web service Version 6.0.14
(C) 2007-2019 Oracle Corporation
All rights reserved.
00:00:00.004219 main     VirtualBox web service 6.0.14 r133895 linux.amd64 (Oct 14 2019 15:00:23) release log
00:00:00.004273 main     Log opened 2020-03-30T09:14:28.846552000Z
00:00:00.004284 main     Build Type: release
00:00:00.004313 main     OS Product: Linux
00:00:00.004316 main     OS Release: 4.19.0-6-amd64
00:00:00.004319 main     OS Version: #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11)
00:00:00.004496 main     DMI Product Name: 
00:00:00.004608 main     DMI Product Version: 
00:00:00.004745 main     Host RAM: 2864MB (2.7GB) total, 136MB available
00:00:00.004754 main     Executable: /usr/lib/virtualbox/vboxwebsrv
00:00:00.004755 main     Process ID: 17576
00:00:00.004757 main     Package type: LINUX_64BITS__DEBIAN_10
00:00:00.031123 main     IPC socket path: /tmp/.vbox-vbox-ipc/ipcd
00:00:00.103608 SQPmp    Socket connection successful: host = 127.0.0.1, port = 18083, master socket = 8
00:00:02.937349 SQPmp    Request 1 on socket 9 queued for processing (1 items on Q)
00:00:02.937427 SQPmp    Request 2 on socket 10 queued for processing (2 items on Q)
00:00:02.937464 SQPmp    Request 3 on socket 11 queued for processing (3 items on Q)
00:00:02.937496 SQPmp    Request 4 on socket 12 queued for processing (4 items on Q)
00:00:02.937705 SQW01    New SOAP thread started
00:00:02.938753 SQPmp    Request 5 on socket 13 queued for processing (5 items on Q)
00:00:02.938862 SQW01    Processing connection from IP=127.0.0.1 socket=9 (0 out of 1 threads idle)
00:00:02.939483 SQPmp    Request 6 on socket 14 queued for processing (5 items on Q)
00:00:02.939869 SQW02    New SOAP thread started
00:00:02.940213 SQW02    Processing connection from IP=127.0.0.1 socket=10 (0 out of 2 threads idle)
00:00:02.952205 SQW02    External authentication library is 'VBoxAuth'
00:00:03.125984 SQW02    Access for user 'vbox' granted
00:00:03.279011 SQW01    Access for user 'vbox' granted
00:00:03.341692 SQW01    Processing connection from IP=127.0.0.1 socket=11 (0 out of 2 threads idle)
00:00:03.347340 SQPmp    Request 7 on socket 16 queued for processing (4 items on Q)
00:00:03.352958 SQW03    New SOAP thread started
00:00:03.353300 SQW03    Processing connection from IP=127.0.0.1 socket=12 (0 out of 3 threads idle)
00:00:03.550665 SQW01    Access for user 'vbox' granted
00:00:03.726861 SQW03    Access for user 'vbox' granted
00:00:03.735693 SQW01    Processing connection from IP=127.0.0.1 socket=13 (0 out of 3 threads idle)
00:00:04.060652 SQW01    Access for user 'vbox' granted
00:00:04.164297 SQW01    Processing connection from IP=127.0.0.1 socket=14 (0 out of 3 threads idle)
00:00:04.400104 SQW01    Access for user 'vbox' granted
00:00:04.427370 SQW01    Processing connection from IP=127.0.0.1 socket=16 (0 out of 3 threads idle)
00:00:04.639614 SQW01    Access for user 'vbox' granted
00:00:05.088789 Watchdog Statistics: 1 websessions, 10 references`

**Comment by PatiB87** [comment]
The message in my phpvirtualbox says:

> There was an error obtaining the list of registered virtual machines from VirtualBox. Make sure vboxwebsrv is running and that the settings in config.php are correct.

The list of virtual machines will not begin auto-refreshing again until this page is reloaded.

HTTP error: 500 Internal Server Error

and when I click "details", I get:

> readyState: "4"
responseText: ""
status: "500"
statusText: "Internal Server Error"

**Comment by PatiB87** [comment]
I also checked the config.php as the error states and compared it to the example: It is literally the same as the example except for the password for the user (which is correct - I tested it).

I don't know where else to start debugging. Is there a more informative log for phpVirtualBox which clearly shows the problem instead of this web interface error message?

**Comment by trasherdk** [comment]
> here was an error obtaining the list of registered virtual machines from VirtualBox.

That error is typically when your browser looses connection to your web-server.
Click OK, and click on the background, and it should refresh.

This assumes that you have a working setup already :dancer:

**Comment by PatiB87** [comment]
Yeah, well, I tried. Clicking OK or reloading the page does not provide any different result. :/

**Comment by trasherdk** [comment]
Well, you know the problem is between your web-server and `vboxwebsrv`, right?

I have a log-file in: `/var/log/virtualbox/vboxweb.log`. That's where I'd start to look...
Next, web-server log-files: `/var/log/httpd/error_log`
Then, if it's a `virtualhost`, those log-files..

Let's see what you find..

**Comment by trasherdk** [comment]
Actually, the error:
```
readyState: "4"
responseText: ""
status: "500"
statusText: "Internal Server Error"
```
Indicates a problem with a `xmlhttprequest` call.
Check your browsers dev-tools to monitor `XHR` requests.

**Comment by choldimir** [comment]
Go to Web Server config and change PHP 7.4 to 8.1. Now, Virtualbox running for me.

**Comment by marsexpress4711** [comment]
@choldimir 
where exactly did you find the Web Server config for phpvirtualbox?
